File: Python/Valid Sudoku.py
class Solution:
    # @param board, a 9x9 2D array
    # @return a boolean
    def isValidSudoku(self, board):
        for i in range(9):
            a = [0] * 9
            b = [0] * 9
            c = [0] * 9
            for j in range(9):
                if board[i][j] != '.':
                    num1 = int(board[i][j]) -1
                    if a[num1]:
                        return False
                    else:
                        a[num1] = 1
                if board[j][i] != '.':
                    num2 = int(board[j][i]) - 1
                    if b[num2]:
                        return False
                    else:
                        b[num2] = 1
                if board[i/3*3+j/3][i%3*3+j%3] != '.':
                    num3 = int(board[i/3*3+j/3][i%3*3+j%3]) -1
                    if  c[num3]:
                        return False
                    else:
                        c[num3] = 1
                
        return True

File: Java/src/Valid Sudoku.java
public class Solution {
    public boolean isValidSudoku(char[][] board){
        for (int i = 0 ; i<9; i++){
            int[] a = new int[9];
            int[] b = new int[9];
            int[] c = new int[9];
            for (int j= 0 ; j<9 ; j++){
                if (board[i][j] != '.'){
                    int num1 =  Character.getNumericValue(board[i][j]) - 1;
                    if (a[num1] != 0){
                        return false;
                    }
                    else{
                        a[num1] = 1;
                    }
                }
                
                if (board[j][i] != '.'){
                    int num2 =  Character.getNumericValue(board[j][i]) - 1;
                    if (b[num2] != 0){
                        return false;
                    }
                    else{
                        b[num2] = 1;
                    }
                }
                
                if (board[i/3*3 + j/3][i%3*3 + j%3] != '.'){
                    int num3 =  Character.getNumericValue(board[i/3*3 + j/3][i%3*3 + j%3]) - 1;
                    if (c[num3] != 0){
                        return false;
                    }
                    else{
                        c[num3] = 1;
                    }
                }

            }
        }
        return true;
    }
}